.github/workflows/airbyte-ci-tests.yml

```diff
@@ -13,34 +13,91 @@ on:
       - synchronize
     paths:
       - airbyte-ci/**
+    paths-ignore:
+      - '**/*.md'
 jobs:
   run-airbyte-ci-tests:
     name: Run Airbyte CI tests
     runs-on: "conn-prod-xlarge-runner"
     steps:
       - name: Checkout Airbyte
         uses: actions/checkout@v3
+      - name: Get changed files
+        uses: dorny/paths-filter@v2
+        id: changes
+        with:
+          filters: |
+            ops:
+              - 'airbyte-ci/connectors/connector_ops/**'
+            base_images:
+              - 'airbyte-ci/connectors/connector_ops/**'
+              - 'airbyte-ci/connectors/base_images/**'
+            pipelines:
+              - 'airbyte-ci/connectors/connector_ops/**'
+              - 'airbyte-ci/connectors/base_images/**'
+              - 'airbyte-ci/connectors/pipelines/**'
+            metadata-lib:
+              - 'airbyte-ci/connectors/metadata/lib/**'
+            metadata-orchestrator:
+              - 'airbyte-ci/connectors/metadata/lib/**'
+              - 'airbyte-ci/connectors/metadata/orchestrator/**'
```

Comment:
I think that ideally this logic should live in the `airbyte_ci` directly:
The `airbyte-ci` click context has the list of modified files. 
I would suggest a small change in the `airbyte-ci test` command to dynamically test packages according to the list of modified files, and skip execution when  and the modified file do not match the filter.

I'll let you decide if this is a change to consider right now.
The benefit is potential consolidation of such a common logic in `airbyte-ci`, the drawback is that it will be slightly slower as on any `airbyte-ci/*` change a dagger job will have start even if no package is touched.

Reply:
Good callout, I like the idea but I think not yet.

Why 👇 

I used to be on the fence of "should airbyte-ci replace github actions?" or "should airbyte-ci be used by github actions?"

Now I'm of the opinion "airbyte-ci be used by github actions" this is because weve been shown that we dont have the resources to support the building and maintenance of a tool of that scope.

For example to successfully replace GHA as the "should it run" decider we need to
1. Have a glob include exclude feature
2. Have it abstracted in a way thats declarative and non cumbersome for an average dev to edit
3. Have the ability to run a `airbyte-ci ci check-all` so that we dont have to spin up dagger multiple times
    1. a warm cache would be useful here as well
    2. and the ability in our abstractions for any airbyte-ci command to call others
6. Improve the github status check reporter so that it can report a skipped status before an in progress
7. Have the development resources available to support the bugs and problems we would see early on building all this out. (Code we write/own always costs more than yaml we dont)

Thats not to say I am against `airbyte-ci` owning this. I just think its a not yet. And that not yet is based on resourcing, priority and evolving our abstractions a little further

Reply:
Sounds good to me!